pkg/pipeline/extract/aggregate/aggregate.go

```diff
@@ -179,10 +182,13 @@ func (aggregate Aggregate) GetMetrics() []config.GenericMap {
 			"by":                                 strings.Join(aggregate.Definition.By, ","),
 			"aggregate":                          string(group.normalizedValues),
 			"value":                              fmt.Sprintf("%f", group.value),
+			"recentRawValues":                    group.RecentRawValues,
 			"count":                              fmt.Sprintf("%d", group.count),
 			aggregate.Definition.Name + "_value": fmt.Sprintf("%f", group.value),
 			strings.Join(aggregate.Definition.By, "_"): string(group.normalizedValues),
 		})
+		// Once reported, we reset the raw values accumulation
+		group.RecentRawValues = make([]float64, 0)
```

Comment:
@ronensc Reminder- reminder ::: I think that we do not have a problem with synchronization ... but you had thoughts that we might ... do you think we should add a semaphore to protect `RecentRawValues` ... actually thinking on this ... when @KalmanMeth will change the pipeline to all channels this will be totally not needed so maybe the only thing I can ask is that you will try to **create a test** to expose a potential race problem ... so we will know that it happens.

Reply:
@eranra Looking at how the code is implemented currently, I don't see a synchronization problem because all the interaction with `RecentRawValues` is done in the same goroutine/thread.
Anyway, I'll try to add a test that checks that it is populated correctly.